0202-Happy-Number.py

```diff
@@ -0,0 +1,25 @@
+'''
+Write an algorithm to determine if a number is "happy".
+
+A happy number is a number defined by the following process: Starting with any positive integer, replace the number by the sum of the squares of its digits, and repeat the process until the number equals 1 (where it will stay), or it loops endlessly in a cycle which does not include 1. Those numbers for which this process ends in 1 are happy numbers.
+
+Example: 
+
+Input: 19
+Output: true
+Explanation: 
+12 + 92 = 82
+82 + 22 = 68
+62 + 82 = 100
+12 + 02 + 02 = 1
+'''
+class Solution:
+    def isHappy(self, n: int) -> bool:
+        prev = set()
+        while n not in prev:
+            prev.add(n)
+            summ = 0
+            for x in str(n):
+                summ += int(x)**2
```

Comment:
Converting numbers into a string and then to int is never a good thing to do. Always avoid converting a number to a string

Reply:
okay